compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codemodifier/HttpPayloadParamIdentifier.java

```diff
@@ -0,0 +1,271 @@
+            }
+        }
+    }
+
+    void validateResource(SyntaxNodeAnalysisContext ctx, FunctionDefinitionNode member, int serviceId) {
+        extractInputParamTypeAndValidate(ctx, member, serviceId);
+    }
+
+    void extractInputParamTypeAndValidate(SyntaxNodeAnalysisContext ctx, FunctionDefinitionNode member,
+                                          int serviceId) {
+
+        Optional<Symbol> resourceMethodSymbolOptional = ctx.semanticModel().symbol(member);
+        int resourceId = member.hashCode();
+        if (resourceMethodSymbolOptional.isEmpty()) {
+            return;
+        }
+        Optional<String> resourceMethodOptional = resourceMethodSymbolOptional.get().getName();
+
+        if (resourceMethodOptional.isPresent()) {
+            String accessor = resourceMethodOptional.get();
+            if (accessor.equals(GET) || accessor.equals(HEAD) || accessor.equals(OPTIONS)) {
+                return; // No modification is done for non-entity body resources
+            }
+        } else {
+            return; // No modification is done for non resources functions
+        }
+
+        Optional<List<ParameterSymbol>> parametersOptional =
+                ((ResourceMethodSymbol) resourceMethodSymbolOptional.get()).typeDescriptor().params();
+        if (parametersOptional.isEmpty()) {
+            return; // No modification is done for non param resources functions
+        }
+
+        List<ParamData> nonAnnotatedParams = new ArrayList<>();
+        List<ParamData> annotatedParams = new ArrayList<>();
+        ParamAvailability paramAvailability = new ParamAvailability();
+        int index = 0;
+        for (ParameterSymbol param : parametersOptional.get()) {
+            List<AnnotationSymbol> annotations = param.annotations().stream()
+                    .filter(annotationSymbol -> annotationSymbol.typeDescriptor().isPresent())
+                    .collect(Collectors.toList());
+            if (annotations.isEmpty()) {
+                nonAnnotatedParams.add(new ParamData(param, index++));
+            } else {
+                annotatedParams.add(new ParamData(param, index++));
+            }
+        }
+
+        for (ParamData annotatedParam : annotatedParams) {
+            validateAnnotatedParams(annotatedParam.getParameterSymbol(), paramAvailability);
+            if (paramAvailability.isAnnotatedPayloadParam()) {
+                return;
+            }
+        }
+
+        for (ParamData nonAnnotatedParam : nonAnnotatedParams) {
+            ParameterSymbol parameterSymbol = nonAnnotatedParam.getParameterSymbol();
+            if (validateNonAnnotatedParams(ctx, parameterSymbol.typeDescriptor(),
+                                           paramAvailability)) { //TODO change name of the func
+                paramAvailability.setPayloadParamSymbol(parameterSymbol);
+                this.payloadParamContextMap.put(ctx.documentId(),
```

Comment:
Using the same key will overwrite the values, so if we have multiple resources in a service this will effectively change only one resource parameter